inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -837,84 +834,92 @@ void ReorderConcatInputsPass::run() {
     }
     int numOfLinkLayers = 0;
 
-    for (auto & l : *pLayers) {
-        // 1st stage locate concat align filter
+    for (auto& l : *pLayers) {
+        // 1st stage locate concat
         LayerInfo info(l);
-        if (!info.isConcatAlignFilter()) {
+        if (!info.isConcat()) {
             continue;
         }
 
-        // 2rd locating concat
-        if (l->outData.size() != 1) {
-            THROW_GNA_EXCEPTION << "no concat layer after concat aligning layer" << l->name;
+        // 2nd stage locate first input in concat
```

Comment:
lets restrict to fp32 only, so >= 2 can work in that way , integer i would not declare as this mode supported

Reply:
I have pushed a fix for it. please review

Reply:
please review the latest changes, I proposed a fix for scale factor propagation